Styleguide: mise à jour du breadcrumb et de la navigation mobile

alinekeller opened the pull request:
**Breadcrumb**
- Ajout des dropdowns qui contiennent les pages de même niveau;
- Sur desktop: toutes les étapes sont affichées, si nécessaire le breadcrumb passe sur plusieurs lignes;
- Sur mobile / tablette: par défaut, uniquement la maison et la dernière étape sont affichées. Entre deux se trouve un bouton avec trois petit points; lorsqu'on clique dessus, le chemin entier est affiché (sur plusieurs lignes si nécessaire).

**Navigation mobile**
Le menu mobile comprend désormais deux navigations:
- Le menu principal, affiché de manière plus simple, sur deux lignes;
- La navigation latérale (nav aside) qui contient les pages de même niveau que la page courante, ainsi que ses pages enfants.

Pour un exemple complet du breadcrumb et de la navigation mobile, voir le modèle de page **Lab Homepage**.

alinekeller committed:
Add link classes to base breadcrumb template

alinekeller committed:
Create dropdown variant

alinekeller committed:
Add breadcrumb dropdown menus

alinekeller committed:
Add "expand breadcrumb button" for small devices

alinekeller committed:
Fix styleint order

alinekeller committed:
Remove unused breadcrumb js function

alinekeller committed:
Add 'expand breadcrumb' button to tagged breadcrumb

alinekeller committed:
Update "breadcrumb tagged"

alinekeller committed:
Create Nav Mobile component

alinekeller committed:
Base: remove "browse" button (nav-toggle)

alinekeller committed:
Base: replace 'nav-main' with 'nav-mobile'

alinekeller committed:
Base: move 'nav-aside' inside 'nav-mobile'

alinekeller committed:
Update side nav structure on Lab demo page

alinekeller committed:
Base: add condition to display regular or tagged breadcrumb

alinekeller committed:
Update page templates

alinekeller committed:
Fix CSS order

alinekeller committed:
Mobile nav aside fine tunning

alinekeller committed:
Fix styleint css order

williambelle commented on assets/components/pages/study-plan/study-plan.twig:
```diff
@@ -23,12 +23,34 @@
+          </button>
+          <ul class="dropdown-menu" id="dropdown-menu-1">
+            <li class="dropdown-item"><a href="#">À propos</a></li>
+            <li class="dropdown-item" current-menu-item-parent"><a href="#">Éducation</a></li>
```
Il y a un `"` en trop.

williambelle commented on assets/components/pages/study-plan-course/study-plan-course.twig:
```diff
@@ -24,14 +24,36 @@
+          </button>
+          <ul class="dropdown-menu" id="dropdown-menu-1">
+            <li class="dropdown-item"><a href="#">À propos</a></li>
+            <li class="dropdown-item" current-menu-item-parent"><a href="#">Éducation</a></li>
```
Il y a un `"` en trop.

williambelle commented on assets/components/molecules/breadcrumb/breadcrumb.scss:
```diff
@@ -149,6 +139,115 @@
+
+// Dropdown
+
+.breadcrumb .dropdown {
```
Si le bouton dropdown se retrouve dans la partie droite du fil d'ariane, c'est impossible de cliquer dessus :

https://github.com/epfl-si/elements/blob/83bce322b78bc569183894dc406fca449bfce5f3/assets/components/molecules/breadcrumb/breadcrumb.scss#L8-L17

![Screenshot from 2024-05-06 16-02-46](https://github.com/epfl-si/elements/assets/2843501/22b7f83c-a447-42a5-8aa6-83e224aa4b83)

williambelle commented on assets/components/molecules/breadcrumb/breadcrumb.scss:
```diff
@@ -149,6 +139,115 @@
+  }
+}
+
+.breadcrumb .dropdown-menu {
```
Dans certains cas, le dropdown n'est pas complètement visible :

![localhost_3000_(Nexus 5X)](https://github.com/epfl-si/elements/assets/2843501/8bb50e0e-fb3e-487c-8fe6-671f3ecf4200)

alinekeller replied:
Ce point là est assez délicat et nécessite un peu de js. J'ai tenté la chose suivante: au clic sur le bouton, on calcule la position du bouton et la largeur du dropdown. Si le dropdown est plus large que l'espace disponible à droite du bouton, on ajoute une classe qui permet de décaler le dropdown sur la gauche. J'ai testé plusieurs cas de figure, et ça a l'air de fonctionner comme voulu.

(à voir si le javascript peut être amélioré, ce n'est pas du tout mon domaine d'expertise)

Seul cas qui reste problématique: quand le breadcrumb item passe sur deux lignes, le dropdown le recouvre partiellement (et recouvre le bouton). On peut quand même le fermer en cliquant n'importe où en dehors du dropdown. Et idéalement, il ne faudrait pas avoir des éléments aussi longs dans un breadcrumb…

![Capture d’écran 2024-05-16 à 18 12 44](https://github.com/epfl-si/elements/assets/6631947/be1477c1-c052-4542-b6c1-e6fbe193bdc5)

williambelle replied:
Pour moi, c'est tout bon comme ça. (j'ai juste formaté le js).

Pour le comportement sur deux lignes, il faut confirmer avec l'équipe WordPress si c'est ok pour eux.

xentenza replied:
Il y a effectivement des choses qui pourraient être plus jolies le comportement sur deux lignes, comme le sous-menu laissant apparaître le bouton, ou encore les éléments s'affichant comme un texte suivi (cf. ci-dessous, mais je ne vois pas comment faire avec un `<li>`).

Mais là, on parle du 1% du 99% du travail déjà effectué. Je valide la PR, bon boulot!

![comportement-breadcrumb](https://github.com/epfl-si/elements/assets/41774881/1e110a29-e06d-46e5-80f5-0c3bc4220718)

williambelle commented on assets/components/organisms/nav-aside/nav-aside.scss:
```diff
@@ -67,7 +85,7 @@
 
     &.active:not(.menu-item-has-children)+li>a:before {
-      content: none;
+      //content: none;
```
:warning:

williambelle commented on assets/components/organisms/nav-aside/nav-aside.scss:
```diff
@@ -25,6 +25,24 @@
+        padding: 0;
+      }
+      a {
+        padding-inline: 1rem;
```
La hiérarchie n'est plus visible dans le menu aside (en dessous de 1200px).

|  > 1200px | < 1200px  | 
|---|---|
| ![Screenshot from 2024-05-07 11-16-23](https://github.com/epfl-si/elements/assets/2843501/6d8d2945-3ecc-4217-a0f1-08ec427328c6) |  ![Screenshot from 2024-05-07 11-18-40](https://github.com/epfl-si/elements/assets/2843501/25b4c482-5d3e-4f2b-a575-f66458f2f222)|
| ![Screenshot from 2024-05-07 10-57-48](https://github.com/epfl-si/elements/assets/2843501/0860d0f1-4bdc-4c40-853d-b7fc4cc1f45f) | ![Screenshot from 2024-05-07 10-59-49](https://github.com/epfl-si/elements/assets/2843501/1d57dcb4-5558-4a77-9466-bd619b78d341) |

alinekeller committed:
Fix dropdown item classes (study plan templates)

alinekeller committed:
Remove breadcrumb container right fading

alinekeller committed:
Improve dropdown position on small devices

The problem: when the breadcrumb item is too close to the right side of the window, the dropdown might be partially hidden outside of the viewport.

The solution: if the width of the dropdown is greater than the space between the button and the right side of the viewport, add class "open-left" to ".dropdown-menu" and use it to change the position of the dropdown.

alinekeller committed:
Bring hierarchy back to mobile side nav

alinekeller committed:
Fix stylelint issue

williambelle committed:
Format js
